skyportal/handlers/api/analysis.py

```diff
@@ -0,0 +1,423 @@
+    AnalysisService,
+    Group,
+)
+
+log = make_log('app/analysis')
+
+_, cfg = load_env()
+
+
+def valid_url(trial_url):
+    """
+    determine if the URL is valid
+    """
+    try:
+        rez = urlparse(trial_url)
+        return all([rez.scheme, rez.netloc])
+    except ValueError:
+        return False
+
+
+class AnalysisServiceHandler(BaseHandler):
+    """Handler for analysis services."""
+
+    @permissions(["Manage Analysis Services"])
+    def post(self):
+        f"""
+        ---
+        description: POST a new Analysis Service.
+        tags:
+          - analysis_services
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  name:
+                    type: string
+                    description: Unique name/identifier of the analysis service.
+                  display_name:
+                    type: string
+                    description: Display name of the analysis service.
+                  description:
+                    type: string
+                    description: Description of the analysis service.
+                  version:
+                    type: string
+                    description: Semantic version (or githash) of the analysis service.
+                  contact_name:
+                    type: string
+                    description: |
+                        Name of person responsible for the service (ie. the maintainer).
+                        This person does not need to be part of this SkyPortal instance.
+                  contact_email:
+                    type: string
+                    description: Email address of the person responsible for the service.
+                  url:
+                    type: string
+                    description: |
+                        URL to running service accessible to this SkyPortal instance.
+                        For example, http://localhost:5000/analysis/<service_name>.
+                  authentication_type:
+                    type: string
+                    description: |
+                        Service authentiction method. One of: {', '.join(f"'{t}'" for t in AUTHENTICATION_TYPES)}.
+                        See https://docs.python-requests.org/en/master/user/authentication/
+                  _authinfo:
+                    type: object
+                    description: |
+                        Authentication secrets for the service. Not needed if authentication_type is "none".
+                        This should be a string that can be parsed by the python json.loads() function and
+                        should contain the key `authentication_type`. Values of this key will be used
+                        to POST into the remote analysis service.
+                  enabled:
+                    type: boolean
+                    description: Whether the service is enabled or not.
+                  type:
```

Comment:
maybe this should be analysis_type? type is pretty generic.

Reply:
Done.